mne/stats/regression.py

```diff
@@ -310,6 +311,9 @@ def solver(X, Y):
 
         # this is the magical part (thanks to Marijn van Vliet):
         # use toeplitz to construct series of diagonals
+        # XXX this is magical but terribly inefficient as
+        # linalg.toeplitz(samples, lags) allocates a dense matrix with
+        # n_samples rows for each condition in the loop !!!
         X[:, cum:cum + tmax_ - tmin_] = linalg.toeplitz(samples, lags)
```

Comment:
linalg.toeplitz(samples, lags)

is terrible in terms of memory allocation.

we need to write our custom function to build a sparse toeplitz matrix and store X as sparse. The density of X in the example is 0.1%